### PR TITLE
Delayed safehome

### PR DESCRIPTION
**Pull request by tonyyng**
The following changes are in Safehomes v2:

- The safehome does not replace the arming location until it is needed. Also, if you cancel RTH or recover from RX failsafe, the original arming location is restored as the home location.
- New safehome_usage_mode option:
  - OFF - Safehomes are not used
  - RTH (default) - Safehomes will be used for both manual RTH and RTH RX failsafe
  - RTH_FS - Safehomes will be used only for RX failsafe
- The Safehome usage mode can be set using the OSD menu
- New OSD message when RTH to safehome is in progress: "DIVERTING TO SAFEHOME"

By not replacing the arming location with the safehome until it is needed, it fixes the problems where the OSD elements involving home (distance, direction, etc) are those of the safehome instead of the arming location.

Also there are some potential safety checks that use "distance_to_home" that won't work as expected when the arming point has been replaced by the safehome.

Resolves these issues:
#6541 Safehome only use in FS state
#6617 Display telemetry from arming location, not safehome

### CODE REVIEW

**Comment by Jetrell** [comment]
@digitalentity @DzikuVx Any chance of this being added to the 2.7 milestone, and merged?
It works great and makes SafeHome more versatile.

File: docs/Settings.md [hunk 1]
```diff
@@ -444,6 +444,7 @@
 | rx_spi_protocol |  |  |
 | rx_spi_rf_channel_count |  |  |
 | safehome_max_distance | 20000 | In order for a safehome to be used, it must be less than this distance (in cm) from the arming point. |
+| safehome_usage_mode | RTH | Used to control when safehomes will be used. |
```
Comment (avsaase) [inline docs/Settings.md]: I suggest adding a brief explanation of the available modes here.

Comment (tonyyng) [reply docs/Settings.md]: Agreed.  Done

**Comment by avsaase** [comment]
I tried this today with default setting and noticed that once the plane reached the safe home position the system messages displayed "diverting to safehome" and "loitering around home". I think it would be better to create one message "loitering around safehome" (if that fits on the osd, haven't checked).

**Comment by tonyyng** [comment]
> I tried this today with default setting and noticed that once the plane reached the safe home position the system messages displayed "diverting to safehome" and "loitering around home". I think it would be better to create one message "loitering around safehome" (if that fits on the osd, haven't checked).

That is a good suggestion.  I've made the changes.  Can you give it another try?  It's still not flying season here.

**Comment by digitalentity** [comment]
Can you please rebase so we can merge this?

**Comment by tonyyng** [comment]
> Can you please rebase so we can merge this?

I don't know how to do this.  Is it documented somewhere?

**Comment by avsaase** [comment]
> > Can you please rebase so we can merge this?
> 
> I don't know how to do this. Is it documented somewhere?

The easiest is probably to merge the current master branch into your branch and resolve all the conflicts.

**Comment by tonyyng** [comment]
> > > Can you please rebase so we can merge this?
> > 
> > 
> > I don't know how to do this. Is it documented somewhere?
> 
> The easiest is probably to merge the current master branch into your branch and resolve all the conflicts.

So:

```tony@ubinav:~/inav$ git branch
* delayed_safehome
  master
  release_2.6.0
  release_2.6.1
  safe_home_test_2.6.1
  safehome_off_warning
tony@ubinav:~/inav$ git checkout delayed_safehome
Switched to branch 'delayed_safehome'
Your branch is up to date with 'origin/delayed_safehome'.
tony@ubinav:~/inav$ git merge master  <- is this right?
```

**Comment by avsaase** [comment]
Yes, that should be correct.

**Comment by avsaase** [comment]
@tonyyng I have resolved the merge conflicts. Is it okay if I push them to your branch?

**Comment by tonyyng** [comment]
> @tonyyng I have resolved the merge conflicts. Is it okay if I push them to your branch?

Sure

**Comment by Mateyhv** [comment]
Great job @tonyying, very useful PR! Not a big deal but I found the options a bit misleading but you might as well have some other considerations in mind of course. To me 'OFF' is ok, 'RTH' I undertsand only as the manual RTH and RTH_FS I see as both RTH + FS. Wouldn't be better RTH_FS (default, all) and FS_ONLY just for FS?

**Comment by tonyyng** [comment]
> Great job @tonyying, very useful PR! Not a big deal but I found the options a bit misleading but you might as well have some other considerations in mind of course. To me 'OFF' is ok, 'RTH' I undertsand only as the manual RTH and RTH_FS I see as both RTH + FS. Wouldn't be better RTH_FS (default, all) and FS_ONLY just for FS?

Maybe, but as long as the behavior is documented accurately, I don't think it makes a difference.

**Pull request merged**